src/gui/qgsnewdatabasetablenamewidget.cpp

```diff
@@ -0,0 +1,266 @@
+/***************************************************************************
+  qgsnewdatabasetablenamewidget.cpp - QgsNewDatabaseTableNameWidget
+
+ ---------------------
+ begin                : 9.3.2020
+ copyright            : (C) 2020 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QTreeWidgetItemIterator>
+
+#include "qgsnewdatabasetablenamewidget.h"
+#include "qgsapplication.h"
+#include "qgsdataitemproviderregistry.h"
+#include "qgsdataitemprovider.h"
+#include "qgsproviderregistry.h"
+#include "qgsprovidermetadata.h"
+
+QgsNewDatabaseTableNameWidget::QgsNewDatabaseTableNameWidget(
+  QgsBrowserGuiModel *browserModel,
+  const QStringList &providersFilter,
+  QWidget *parent )
+  : QWidget( parent )
+{
+
+  // Initalize the browser
+  if ( ! browserModel )
+  {
+    mBrowserModel = new QgsBrowserGuiModel( this );
+    mBrowserModel->initialize();
+  }
+  else
+  {
+    mBrowserModel = browserModel;
+    mBrowserModel->initialize();
+  }
+
+  setupUi( this );
+
+  mValidationResults->setStyleSheet( QStringLiteral( "* { font-weight: bold; color: red; }" ) );
+
+  QStringList hiddenProviders
```

Comment:
Hm, this is not great. I can easily see this getting out of sync with the existing providers. How would you feel about adding a provider "whitelist" to QgsBrowserProxyModel instead?

Reply:
That makes sense. I would also rename setDataItemProviderKeyFilter to setDataItemProviderKeyExcludeFilter and create a new setDataItemProviderKeyIncludeFilter

Reply:
BTW, keep in mind that we would need to create a white or a black list in any case: we cannot show unsupported root items in the browser tree widget.